### PR TITLE
update config for exciting mirageos3 developments

### PR DESCRIPTION
**Pull request by yomimono**

### CODE REVIEW

File: src/config.ml [hunk 1]
```diff
@@ -77,16 +77,15 @@ let tmplfs = generic_kv_ro ~key:fs_key "../tmpl"
...
 
 let http =
   foreign ~keys "Dispatch.Make"
     (http @-> kv_ro @-> kv_ro @-> pclock @-> job)
 
 let https =
-  let libraries = [ "tls"; "tls.mirage"; "mirage-http" ] in
-  let packages = ["tls"; "tls"; "mirage-http"] in
-  foreign ~libraries ~packages  ~keys "Dispatch_tls.Make"
+  let packages = [package "tls"; package "tls"; package "mirage-http"] in
```
Comment (Drup) [inline src/config.ml]: tls twice ?

**Comment by yomimono** [comment]
I'm puzzled by the failures at https://travis-ci.org/mirage/mirage-www/jobs/181546898 and https://travis-ci.org/mirage/mirage-www/jobs/181546899 .  The build stalls in the `cd src && make depend` phase, but I can't reproduce this locally.

**Comment by yomimono** [comment]
I think this may be an unintended consequence of the pinning workflow -- we're setting keys which are added to the filename of the generated opam file which have `.` in them, which might be interpreted as version numbers -- see `opam pin add --no-action --yes mirage-unikernel-mirage.io-xen .`

**Comment by yomimono** [comment]
PR for the above at https://github.com/mirage/functoria/pull/89 .  Pinning to the fixed branch solves this locally.

**Comment by yomimono** [comment]
The TLS deploy build now fails because `mirage` no longer recognizes `../tls` as a valid argument to `generic_kv_ro` when the `kv_ro` that will be chosen is `fat` (or at least, not `crunch`).

**Comment by yomimono** [comment]
Patch for `invalid_segment` at https://github.com/mirage/mirage/pull/723 but we're still not done; we get a main.ml that doesn't generate a `module Block = ` but tries to reference one.

**Comment by yomimono** [comment]
Further fixes added in mirage/mirage#723 ; this now at least builds locally.  We'll see what Travis thinks once those changes are merged.

**Comment by yomimono** [comment]
At long last our hearts are glad again.

**Pull request merged**